homeassistant/components/ziggo_next/media_player.py

```diff
@@ -0,0 +1,183 @@
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.const import (
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+    STATE_UNAVAILABLE,
+)
+
+from .const import DOMAIN, ZIGGO_API
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up entry."""
+    players = []
+    api = hass.data[ZIGGO_API]
+    for box_id in api.settopBoxes.keys():
+        box = api.settopBoxes[box_id]
+        players.append(ZiggoNextMediaPlayer(box_id, box.name, api))
+    async_add_devices(players, update_before_add=True)
+
+
+class ZiggoNextMediaPlayer(MediaPlayerDevice):
+    """The home assistant media player."""
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.box_id)
+            },
+            "name": self.box_name,
+            "manufacturer": "Ziggo",
+            "model": "Mediabox Next",
+        }
+
+    def __init__(self, box_id: str, name: str, api: ZiggoNext):
+        """Init the media player."""
+        self.api = api
+        self.box_id = box_id
+        self.box_name = name
+        self.box_state = None
+        self.box_info = None
+
+    def update(self):
+        """Update the box."""
+        self.api.load_channels()
+        box = self.api.settopBoxes[self.box_id]
+        self.box_state = box.state
+        self.box_info = box.info
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self.box_name
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        if self.box_state == ONLINE_RUNNING:
+            if self.box_info is not None and self.box_info.paused:
+                return STATE_PAUSED
+            return STATE_PLAYING
+        if self.box_state == ONLINE_STANDBY:
+            return STATE_OFF
+        return STATE_UNAVAILABLE
+
+    @property
+    def media_content_type(self):
+        """Return the media type."""
+        if self.box_info.sourceType == "app":
+            return MEDIA_TYPE_APP
+        return MEDIA_TYPE_CHANNEL
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        if self.box_info.sourceType == "app":
+            return SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+        return (
+            SUPPORT_PLAY
+            | SUPPORT_PAUSE
+            | SUPPORT_TURN_ON
+            | SUPPORT_TURN_OFF
+            | SUPPORT_SELECT_SOURCE
+            | SUPPORT_NEXT_TRACK
+            | SUPPORT_PREVIOUS_TRACK
+        )
+
+    @property
+    def available(self):
+        """Return True if the device is available."""
+        available = self.api.is_available(self.box_id)
```

Comment:
Does this generate I/O? If yes, the fetch should be done inside `update` and stored into a member variable.

Reply:
No it doesn't. The api is updated asynchronous.  This call is_available checks for the last known state.